uqcsbot/scripts/jobs.py

```diff
@@ -21,68 +23,86 @@
     f"If you have any questions, please get in touch with the committee in #uqcs-meta or by email at "
     f"<{UQCS_EMAIL}|contact@uqcs.org.au>."
 ]
 
 
+def replace_channel_links(messages: List[str]) -> List[str]:
+    """
+    Replace static channel names for #uqcs-meta, #jobs-discussion, #jobs-bulletin with links.
+    :param messages: the list of welcome messages to replace links in.
+    :param jobs_bulletin: the link for the #jobs-bulletin channel.
+    :param jobs_discussion: the link for the #jobs-discussion channel.
+    :param uqcs_meta: the link for the #uqcs-meta channel.
+    :return: the same list of strings, but with static channel names replaced with links.
+    """
+    linked_messages = []
+    for message in messages:
+        message = message.replace("#jobs-bulletin", f"<#{bot.channels.get('jobs-bulletin').id}>")
+        message = message.replace("#jobs-discussion", f"<#{bot.channels.get('jobs-discussion').id}>")
+        message = message.replace("#uqcs-meta", f"<#{bot.channels.get('uqcs-meta').id}>")
+        linked_messages.append(message)
+    return linked_messages
+
+
 @bot.on("member_joined_channel")
-def welcome_jobs(evt: dict):
+def welcome_jobs(event: dict):
     """
     Welcomes job seekers and employers to the #jobs-bulletin channel, setting expectations for both.
 
     @no_help
     """
-    chan = bot.channels.get(evt.get("channel"))
+    chan = bot.channels.get(event.get("channel"))
     if chan is None or chan.name != "jobs-bulletin":
         return
 
     jobs_bulletin = chan
-    user = bot.users.get(evt.get("user"))
+    user = bot.users.get(event.get("user"))
 
     if user is None or user.is_bot:
         return
 
     # Send instructions to user
     bot.post_message(user.user_id, f"Hey {user.display_name}, welcome to <#{jobs_bulletin.id}>!")
-    for message in WELCOME_MESSAGES:
+    for message in replace_channel_links(WELCOME_MESSAGES):
         time.sleep(MESSAGE_PAUSE)
         bot.post_message(user.user_id, message)
 
 
 @bot.on("message")
-def job_response(evt: dict):
+def job_response(event: dict):
     """
     Messages users that have posted in #jobs-bulletin to remind them of the rules.
 
     @no_help
     """
-    chan = bot.channels.get(evt.get("channel"))
+    chan = bot.channels.get(event.get("channel"))
 
     if chan.name != "jobs-bulletin":
         return
 
-    if evt.get("subtype") in ["channel_join", "channel_leave"]:
+    if event.get("subtype") in ["channel_join", "channel_leave"]:
         return
 
     jobs_bulletin = chan
     jobs_discussion = bot.channels.get("jobs-discussion")
-    user = bot.users.get(evt.get("user"))
+    user = bot.users.get(event.get("user"))
 
     if user is None or user.is_bot:
         return
 
-    bot.post_message(jobs_bulletin, f"{user.display_name} has posted a new job in <#{jobs_bulletin.id}>! "
+    bot.post_message(jobs_bulletin, f"{user.display_name} has posted a new job! "
```

Comment:
small request, could you fix `display_name` so that it defaults to the user's `real_name`? you'll have to do this in `api.py`. lmk if you need help with that

Reply:
oh yeah, reason is that `display_name` is optional for older accounts (such as mine I believe?)

Reply:
alternatively, you could set `name = display_name or real_name` in the user class of api.py which would make things easier, and then change all usages of it throughout the codebase (where applicable)

Reply:
up to you

Reply:
also, did we want to do a double message here? i.e. send to jobs-discussion too?